APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift

```diff
@@ -199,12 +199,12 @@ private func checkAsyncMethods(purchases: Purchases) async {
 
         let _: [StoreProduct] = await purchases.products([])
         let discount: StoreProductDiscount! = nil
-        let _: (StoreTransaction, CustomerInfo, Bool) = try await purchases.purchase(package: pack)
-        let _: (StoreTransaction, CustomerInfo, Bool) = try await purchases.purchase(package: pack,
-                                                                                     discount: discount)
-        let _: (StoreTransaction, CustomerInfo, Bool) = try await purchases.purchase(product: stp)
-        let _: (StoreTransaction, CustomerInfo, Bool) = try await purchases.purchase(product: stp,
-                                                                                     discount: discount)
+        let _: (StoreTransaction?, CustomerInfo, Bool) = try await purchases.purchase(package: pack)
```

Comment:
This was incorrect before. The only reason it compiled is because we had:
```swift
guard let transaction = maybeTransaction else {
  fatalError("Expected non-nil result 'transaction' for nil error")
}
```
But if the purchase was cancelled, `Product.PurchaseResult` does not provide a `Transaction`.